src/drv_htc_vive/vive.h

```diff
@@ -63,6 +70,23 @@ typedef struct
 	float gyro_range;
 } vive_imu_config;
 
+typedef struct
+{
+	uint8_t id;
+	uint32_t firmware_version;
+	uint32_t unknown1;
+	uint8_t string1[16];
+	uint8_t string2[16];
+	uint8_t hardware_version_micro;
+	uint8_t hardware_version_minor;
+	uint8_t hardware_version_major;
+	uint8_t hardware_revision;
+	uint32_t unknown2;
+	uint8_t fpga_version_minor;
+	uint8_t fpga_version_major;
+	uint8_t reserved[13];
+} __attribute__((packed)) vive_firmware_version_packet;
```

Comment:
As you wrote. is it a good idea? it should break the compilation with MSVC.
Maybe adding a define and including special push/pop file for setting packed on MSVC would be better?

Reply:
@alexandre-janniaux Do we have a CI for MSVC? I don't have a license for that. Would be glad if someone could try to build / run this on other platforms.

Reply:
No we don't. It should be possible with AppVeyor or maybe Travis. I'll take a look if I find some time in the week.

Reply:
Any update on this? we should not push code anymore unless it is confirmed to work on all 3 major compilers (MSVC, Clang and GCC) for the common platforms.

Reply:
I have hacked together an AppVeyor Windows build in #204. Indeed this branch does not build on MSVC:
https://ci.appveyor.com/project/pH5/openhmd/builds/22942046